current/runtime/storage.md

```diff
@@ -4,22 +4,128 @@ lang: en
-* [Storage Double Map](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html) - An implementation of a map with two keys.
+* [Storage Value](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageValue.html) - A single value
+* [Storage Map](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html) - A key-value hash map
+* [Storage Double Map](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html) - An
+  implementation of a map with two keys that provides the important ability to efficiently remove all entries that have
+  a common first key
+* Iterable Storage Maps - Implementations of
+  [Storage Map](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageMap.html) and
+  [Storage Double Map](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageDoubleMap.html)
+  whose keys and values can be iterated over
 
-Any value which can be encoded by the [Parity SCALE codec](../advanced/codec) is supported by these storage APIs.
+The type of Storage Item you select should depend on the logical way in which the value will be used by your runtime.
 
-### Storage Declaration
+### Storage Value
 
-You can use the `decl_storage!` macro to easily create new runtime storage items. Here is an example of what it looks like to declare each type of storage item:
+This type of storage item should be used for values that are viewed as a single unit by the runtime, whether that is a
+single primitive value, a single `struct` or a single collection of related items. Although wrapping related items in
+a shared `struct` is an excellent way to reduce the number of storage reads (an important consideration), at some point
+the size of the object will begin to incur costs that may outweigh the optimization in storage reads. Storage values
+can be used to store lists of items, but runtime developers should take care with respect to the size of these lists.
+Large lists incur storage costs just like large `structs`. Furthermore, iterating over a large list in your runtime may
+result in exceeding the block production time.
+
+#### Methods
+
+Refer to the Storage Value documentation for
+[a comprehensive list of the methods that Storage Values expose](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageValue.html#required-methods).
+Some of the most important methods are summarized here:
+
+##### `get()`
+
+Load the value from storage.
+
+##### `put(val)`
+
+Store the provided value.
+
+##### `mutate(fn)`
+
+Mutate the value with the provided function.
+
+##### `take()`
+
+Remove the value from storage.
+
+### Storage Maps
+
+Storage Maps are implemented as hash maps, which is a pattern that should be familiar to most developers. In order to
```

Comment:
> Storage Maps are implemented as hash maps

maybe it is only me but for me hash map is a data structure which handle collision. at least if presented this way it should be clear that collision are not handled.

But actually instead of using "hash map" term I would rather say that it is a map where key are hashed.

Reply:
Thank you for this clarification. Please let me know if you think my solution was sufficient.

Reply:
hmm I see, I'm not sure actually what I proposed was better, like

> Storage Maps are implemented as maps with hashed keys, which is a pattern that should be familiar to most developers.

wouldn't have make sense to me I think if I didn't know it already, not sure how to do better. (still for me the clearest is that a storage map insert at the path `hash(encode(key))` the value encoded.)

Reply:
Tried to clarify this in the docs :+1: